### PR TITLE
[backend] add index

### PR DESCRIPTION
**Pull request by savacano28**
### Proposed changes

* Add indexes for atomic testing and joins
*

### Related issues

*
*

### Checklist

- [x] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class V3_12__Add_index_for_atomic_testings extends BaseJavaMigration {
+
+  @Override
+  public void migrate(Context context) throws Exception {
+    Connection connection = context.getConnection();
+    Statement select = connection.createStatement();
+    select.execute(""
+        + "CREATE INDEX idx_fk_inject_expectation ON injects_expectations (inject_id);"
+        + "CREATE INDEX idx_fk_inject_tag ON injects_tags (inject_id);"
+        + "CREATE INDEX idx_fk_inject_assets ON injects_assets (inject_id);"
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java:19]: Already exists -> idx_injects_assets_inject

File: openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        + "CREATE INDEX idx_fk_inject_asset_groups ON injects_asset_groups (inject_id);"
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java:20]: Already exists -> idx_injects_asset_groups_inject

File: openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        + "CREATE INDEX idx_fk_inject_teams ON injects_teams (inject_id);"
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java:21]: Already exists -> idx_96e1b96c7983aee

File: openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        + "CREATE INDEX idx_fk_inject_injects_documents ON injects_documents (inject_id);"
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java:22]: Already exists -> idx_injects_documents_inject

File: openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        + "CREATE INDEX idx_null_exercise_and_scenario ON injects (inject_id) WHERE inject_scenario IS NULL AND inject_exercise IS NULL;"
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java:23]: Why we need this index ?

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java (hunk 5 above)
Comment (savacano28) [reply openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java:23]: The last one is for atomic testing, where the condition to filter injects is that the scenario and exercise are null.

Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java]: Okay, but we already have an index on inject_id
How does this add improvement? (sorry but MY knowledge of indexes is limited)

Comment (savacano28) [reply openbas-api/src/main/java/io/openbas/migration/V3_12__Add_index_for_atomic_testings.java]: It depends on the type of search being performed. A search by injectId is different from a search by exercise and scenario... but, if the index is useful, it is the planificator who decides...

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**